hypothesis-python/src/hypothesis/strategies/_internal/types.py

```diff
@@ -256,9 +256,27 @@ def is_a_union(thing):
     )
 
 
+try:
+    import numpy as np
+except ImportError:
+
+    def is_np_type(thing):
+        return False
+
+else:
+
+    def is_np_type(thing):
+        return isinstance(thing, (np.ndarray, np.generic))
+
+
 def is_a_type(thing):
     """Return True if thing is a type or a generic type like thing."""
-    return isinstance(thing, type) or is_generic_type(thing) or is_a_new_type(thing)
+    return (
+        isinstance(thing, type)
+        or is_generic_type(thing)
+        or is_a_new_type(thing)
+        or is_np_type(thing)
```

Comment:
```python-traceback
File "hypothesis/strategies/_internal/core.py", line 1074, in _from_type
    raise InvalidArgument(f"thing={thing!r} must be a type")  # pragma: no cover
hypothesis.errors.InvalidArgument: thing=numpy.ndarray[typing.Any, numpy.dtype[numpy.float64]] must be a type
```
This suggests that `is_np_type()` returns False for `numpy.ndarray[typing.Any, numpy.dtype[numpy.float64]]`!  I don't have suggestions on how to fix that for you, but could take a look if you get stuck.

Reply:
🤔 

OK, I think this is now a fairly cursed problem; I'll try to get to the bottom of it this weekend.  Thanks for taking it so far!

Reply:
To make things worse, I am trying it locally with Python 3.8.13 and the `is_a_type()` returns True. While the CI test used 3.8.14 and it returns False

Reply:
This is the error I got for 3.8.13:
```
  File "/Users/cheuktingho/Contributions/hypothesis/hypothesis-python/src/hypothesis/strategies/_internal/core.py", line 1150, in _from_type
    strategies = [
  File "/Users/cheuktingho/Contributions/hypothesis/hypothesis-python/src/hypothesis/strategies/_internal/core.py", line 1154, in <listcomp>
    and issubclass(k, thing)
  File "/Users/cheuktingho/.pyenv/versions/3.8.13/envs/hypothesis38/lib/python3.8/site-packages/numpy/typing/_generic_alias.py", line 149, in __subclasscheck__
    raise TypeError("issubclass() argument 2 cannot be a "
TypeError: issubclass() argument 2 cannot be a parameterized generic
```

Reply:
> OK, I think this is now a fairly cursed problem; I'll try to get to the bottom of it this weekend. Thanks for taking it so far!

@Zac-HD  I have been doing lots of array-type stuff lately. I can try to take a look at this in the next few days.

Reply:
> To make things worse, I am trying it locally with Python 3.8.13 and the is_a_type() returns True. While the CI test used 3.8.14 and it returns False

Wow that is so gross...

Reply:
Thanks Ryan, and good luck!

Reply:
> It seems that it is only failing in Python 3.8

@Cheukting just a heads-up: the numpy tests are only run against Python 3.8 on our CI currently